src/nvim/decoration.c

```diff
@@ -171,28 +172,29 @@ bool decor_redraw_start(buf_T *buf, int top_row, DecorState *state)
     uint64_t start_id = mark.id & ~MARKTREE_END_FLAG;
     ExtmarkItem *item = map_ref(uint64_t, ExtmarkItem)(buf->b_extmark_index,
                                                        start_id, false);
-    if (!item || !item->decor) {
+    if (!item) {
```

Comment:
with my previous suggestion, this need not be changed, which is important for performance (we want to skip over all non-decoration extmarks in screen.c very quickly, in the future perhaps even filtering them in the marktree structure itself).

Reply:
done